zerver/views/users.py

```diff
@@ -269,7 +277,7 @@ def update_user_backend(
     ),
 ) -> HttpResponse:
     should_notify_user_on_updated_profile = False
-    old_full_name = None
+    old_full_name, old_role, new_role = None, None, None
```

Comment:
I think we can maintain only two variables `old_value` and `new_value`.

And define a parameter in `notify_users_on_updated_user_profile` named `attribute_changed` or something better. So, finally we'll have:
```python
def notify_users_on_updated_user_profile(
    acting_user: UserProfile,
    recipient_user: UserProfile,
    
    *,
    attributed_changed: ,
    old_value: ,
    new_value: ,
) 
```
`attribute_changed` can have values like **role**, **full_name**, etc